server/src/api/post.rs

```diff
@@ -719,7 +724,265 @@ impl Perform for Oper<EditPost> {
 
     if let Some(ws) = websocket_info {
       ws.chatserver.do_send(SendPost {
-        op: UserOperation::EditPost,
+        op: UserOperation::DeletePost,
+        post: res.clone(),
+        my_id: ws.id,
+      });
+    }
+
+    Ok(res)
+  }
+}
+
+#[async_trait::async_trait(?Send)]
+impl Perform for Oper<RemovePost> {
+  type Response = PostResponse;
+
+  async fn perform(
+    &self,
+    pool: &DbPool,
+    websocket_info: Option<WebsocketInfo>,
+  ) -> Result<PostResponse, LemmyError> {
+    let data: &RemovePost = &self.data;
+
+    let claims = match Claims::decode(&data.auth) {
+      Ok(claims) => claims.claims,
+      Err(_e) => return Err(APIError::err("not_logged_in").into()),
+    };
+
+    let user_id = claims.id;
+
+    let edit_id = data.edit_id;
+    let orig_post = blocking(pool, move |conn| Post::read(conn, edit_id)).await??;
+
+    // Check for a site ban
+    let user = blocking(pool, move |conn| User_::read(conn, user_id)).await??;
+    if user.banned {
+      return Err(APIError::err("site_ban").into());
+    }
+
+    // Check for a community ban
+    let community_id = orig_post.community_id;
+    let is_banned =
+      move |conn: &'_ _| CommunityUserBanView::get(conn, user_id, community_id).is_ok();
+    if blocking(pool, is_banned).await? {
+      return Err(APIError::err("community_ban").into());
+    }
+
+    // Verify that only the mods can remove
+    let mods_and_admins = blocking(pool, move |conn| {
+      Community::community_mods_and_admins(conn, community_id)
+    })
+    .await??;
+    if !mods_and_admins.contains(&user_id) {
+      return Err(APIError::err("not_an_admin").into());
+    }
```

Comment:
`is_mod(user_id, community_id) -> Result<(), SecurityError>`

Reply:
I'll see if this is possible with the blocking call.

Reply:
Okay I found a way to reformulate it like that, but it just adds code. Its too late now tho.